### ISSUE TITLE
'Browserpass' started debugging this browser

### ISSUE THREAD

**Issue by ckotte** [browserpass/browserpass-extension#223]
### General information

-   Operating system + version: ```MacOS Mojave 10.14.6```
-   Browser + version: ```Chrome Version 84.0.4147.105 (Official Build) (64-bit)```
-   Information about the host app:
    - Installed via homebrew
    ```
    # brew info browserpass                                                                                                                                                                                 
    amar1729/formulae/browserpass: stable 3.0.6 (bottled)
    This is a host application for browserpass browser extension providing it access to your password store.
    https://github.com/browserpass/browserpass-native
    /usr/local/Cellar/browserpass/3.0.6 (11 files, 4MB) *
      Poured from bottle on 2019-06-09 at 11:32:45
    From: https://github.com/amar1729/homebrew-formulae/blob/HEAD/browserpass.rb
    ==> Dependencies
    Build: coreutils ✔, gnu-sed ✔
    Required: gpg ✔, pinentry ✔, pinentry-mac ✔
    ==> Caveats
    ********************************************************************************
    * To configure your browser, RUN THE FOLLOWING:
    
    $ PREFIX='/usr/local/opt/browserpass' make hosts-BROWSER-user -f /usr/local/opt/browserpass/lib/browserpass/Makefile
    
    * Where BROWSER is one of the following: [chromium chrome vivaldi brave firefox]
    ********************************************************************************
    ```
-   Information about the browser extension:
    -   Installed via Chrome
    -   Browserpass extension version as reported by your browser: 3.5.0

### Exact steps to reproduce the problem

1. Visit any website

2. Open pass entry with Browserpass

3. Username and password is entered automatically

### What should happen?

No popup message

### What happened instead?

Chrome shows a popup message at the top

![image](https://user-images.githubusercontent.com/8488565/89032564-3b4f4680-d335-11ea-960a-f7170ef2968a.png)

Everything works. I just want to get rid of the popup message.

**Comment by maximbaz** [comment]
Does that message appear for a second and disappear, or it stays there forever until you hit Cancel?

**Comment by ckotte** [comment]
It stays there forever until I hit Cancel or x. Looks like it doesn't make a difference. Browserpass still works afterwards.

**Comment by maximbaz** [comment]
I confirm the issue. I looked into it, sadly seems some sort of regression in Chromium. If I open devtools of the background page of the extension, previously the snippet below would show and hide the bar, now the bar stays open forever.

```js
chrome.debugger.attach({ tabId: 5 }, "1.3", function() { 
   chrome.debugger.detach({ tabId: 5 });
})
```

> Visit any website

Does it happen on any website for you? Debugger is only used when submit button cannot be found, I would be very surprised if you get this message on github.com/login - could you check please?

I don't know if disabling autosubmit is a good workaround ( 😄 ), but unfortunately I dont see what else we can do on our side...

**Comment by ckotte** [comment]
Ok. Sorry. It doesn't happen on any website. It doesn't happen on GitHub and Google.

How do you search for the submit button? Here it cannot find the 'Anmelden' button: https://kunde.comdirect.de/lp/wt/login?execution=e2s1.

Btw. In the settings you don't recommend to use autosubmit. ```Automatically submit forms after filling (not recommended)```

**Comment by maximbaz** [comment]
> Ok. Sorry. It doesn't happen on any website. It doesn't happen on GitHub and Google.

Good, then at least we conclude that debugger isn't triggered more often that it should be...

> How do you search for the submit button? Here it cannot find the 'Anmelden' button: https://kunde.comdirect.de/lp/wt/login?execution=e2s1.

We have a relatively strict way for searching the button, in order to avoid clicking on wrong buttons when your credentials are written on the page, the logic being that it is safer to not click anything (or press Enter in the form using debugger) than to click wrong button.

The full list of accepted buttons is here: 

https://github.com/browserpass/browserpass-extension/blob/585719a698626b3f6e637f9f7d7968e1b03c24a2/src/inject.js#L40-L80

Anmelden is internally rendered as an `<a>` tag, so since it isn't a button at all, we don't click it.

> Btw. In the settings you don't recommend to use autosubmit.

Yep, we have some notes in readme: https://github.com/browserpass/browserpass-extension#a-note-about-autosubmit

I just thought that if you are using autosubmit, it must mean you accepted the risk, and then we try as hard as possible to perform this autosubmit (which includes using debugger api when absolutely necessary).

**Comment by maximbaz** [comment]
I'll close because I think there is unfortunately nothing we can do, and it should automatically get fixed once Chromium fixes the behavior of `chrome.debugger.detach()`, but if anyone has any ideas on what can be done on our side, please do share 🙂

**Comment by erayd** [comment]
> ...once Chromium fixes the behavior of `chrome.debugger.detach()`

Is this actually a bug? It's conceivable that the Chromium folks might have done this deliberately, to ensure the user is notified if an extension is using that API, considering the high level of access that can be obtained by doing so.

**Comment by maximbaz** [comment]
Might be deliberate, but is there anything we can do apart from disabling autosubmit?

**Comment by erayd** [comment]
Unsure at this point. If it's deliberate, we might just have to live with it. But IMO we shouldn't close this issue until we know why it's occurring.

**Comment by ckotte** [comment]
> Yep, we have some notes in readme: https://github.com/browserpass/browserpass-extension#a-note-about-autosubmit

> I just thought that if you are using autosubmit, it must mean you accepted the risk, and then we try as hard as possible to perform this autosubmit (which includes using debugger api when absolutely necessary).

Ok. I must have enabled it some time ago and just forgot about it. I'm using this extension for a long time already.

Thanks for your help and clarification. I will just disable autosubmit.

**Comment by lukyth** [comment]
In case it could help with you folks since I was facing with this issue as well. Apparently, this new behavior was made intentionally into Chromium https://chromium-review.googlesource.com/c/chromium/src/+/2159967

> Remove extension dev tools infobar only when the user closes/cancels it.
>
> Previously, the bar also disappeared if it was no longer applicable,
i.e. no more extensions were attached.  This is problematic since the
extension may have made persistent changes to the page, and the user
needs to be guaranteed an opportunity to see something telling them
that.

So all we could do is just to deal with it I think, or maybe explaining this in an FAQ. 😂

**Comment by maximbaz** [comment]
Nice finding! Thanks! Good point about adding a note in README 👍

**Comment by maximbaz** [comment]
I have received other feedback where this "debugger" popup appeared when a user tried to login to their online banking, and it got them really really worried.

What do you think if we remove debugger permission altogether? Then browserpass will just leave the cursor in the form and let users press Enter, just like it does in Firefox.

**Comment by erayd** [comment]
@maximbaz I'd be more comfortable with removing it I think. If it's going to remain, then we shouldn't be using it, because it'll alarm users (as demonstrated by the feedback you received).
